src/SolutionAcceleration.f90

```diff
@@ -0,0 +1,385 @@
+CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief Initializes the Solution Acceleration base class
+!> @param solver The Solution Acceleration solver to act on
+!> @param ce The computing environment to use for the calculation
+!> @param Params The options parameter list
+!>
+SUBROUTINE init_base_SolutionAccelerationType(solver,ce,Params)
+  CLASS(SolutionAccelerationType),INTENT(INOUT) :: solver
+  TYPE(FutilityComputingEnvironment),TARGET,INTENT(IN) :: ce
+  TYPE(ParamType),INTENT(IN) :: Params
+
+  CHARACTER(LEN=*),PARAMETER :: myName='init_base_SolutionAccelerationType'
+
+  REQUIRE(Params%has('SolutionAcceleration->num_unknowns'))
+  REQUIRE(.NOT.solver%isInit)
+
+  !Pull Data from Parameter List
+  CALL Params%get('SolutionAcceleration->num_unknowns',solver%n)
+  IF(Params%has('SolutionAcceleration->starting_iteration')) &
+      CALL Params%get('SolutionAcceleration->starting_iteration',solver%start)
+
+  IF(solver%n < 1) CALL ce%exceptHandler%raiseError('Incorrect input to '//modName// &
+      '::'//myName//' - Number of unkowns (n) must be greater than 0!')
+
+  IF(solver%start < 1) CALL ce%exceptHandler%raiseError('Incorrect input to '//modName// &
+      '::'//myName//' - Starting point must be greater than 1!')
+
+  solver%s=0
+  solver%ce => ce
+ENDSUBROUTINE init_base_SolutionAccelerationType
+!
+!-------------------------------------------------------------------------------
+!> @brief Clears the solution acceleration base class
+!> @param solver The solver to act on
+!>
+SUBROUTINE clear_base_SolutionAccelerationType(solver)
+    CLASS(SolutionAccelerationType),INTENT(INOUT) :: solver
+
+    INTEGER(SIK) :: i
+
+    IF(ALLOCATED(solver%x)) THEN
+      DO i=1,SIZE(solver%x)
+        CALL solver%x(i)%clear()
+      ENDDO
+      DEALLOCATE(solver%x)
+    ENDIF
+    solver%s=0
+    solver%n=-1
+    solver%start=1
+    solver%isInit=.FALSE.
+  ENDSUBROUTINE clear_base_SolutionAccelerationType
+!
+!-------------------------------------------------------------------------------
+!> @brief Initializes the relaxed picard type
+!> @param solver The relaxed picard solver to act on
+!> @param ce The computing environment to use for the calculation
+!> @param Params The options parameter list
+!>
+SUBROUTINE init_RelaxedPicardType(solver,ce,Params)
+  CLASS(RelaxedPicardType),INTENT(INOUT) :: solver
+  TYPE(FutilityComputingEnvironment),TARGET,INTENT(IN) :: ce
+  TYPE(ParamType),INTENT(IN) :: Params
+
+  CHARACTER(LEN=*),PARAMETER :: myName='init_RelaxedPicardType'
+
+  REQUIRE(.NOT.solver%isInit)
+
+  CALL solver%init_base(ce,Params)
+
+  IF(Params%has('SolutionAcceleration->picard_relaxation_factor')) &
+      CALL Params%get('SolutionAcceleration->picard_relaxation_factor',solver%alpha)
+
+  IF(solver%alpha <= 0.0_SRK) CALL ce%exceptHandler%raiseError('Incorrect input to '//modName// &
+      '::'//myName//' - Relaxation coefficient must be greater than 0!')
```

Comment:
Leaving open the possibility of alpha > 1.0 are we? How devious.

Reply:
yes